8-better-way-utilizing-object-pool-nocanvas-svg/js/index.js

```diff
@@ -27,22 +28,27 @@ Bubble.prototype = {
     this.vx = vx;
     this.vy = vy;
   },
-  update: function () {
-    this.x += this.vx;
-    this.y += this.vy;
-    this.vy += gravity;
-    this.vx = this.vx*friction;
-    this.vy = this.vy*friction;
+  update: function (dt) {
+    // friction opposes the direction of velocity
+    var acceleration = -Math.sign(this.vx) * friction;
+    // distance = velocity * time + 0.5 * acceleration * (time ^ 2)
+    this.x += this.vx * dt + 0.5 * acceleration * (dt ^ 2);
+    this.y += this.vy * dt + 0.5 * gravity * (dt ^ 2);
```

Comment:
@sdras I had to refresh it in my head just recently when I worked on the game https://github.com/w0rm/elm-mogee :)